src/rust/ide/view/graph-editor/src/lib.rs

```diff
@@ -2039,6 +2042,50 @@ impl GraphEditor {
         self.frp.add_node.emit(());
         self.frp.output.node_added.value()
     }
+
+    /// Ads a new node below `parent` and returns its ID. If there is not enough space right below
+    /// `parent` then the new node is moved to the right to first gap that is large enough.
+    pub fn add_node_below(&self, parent:NodeId) -> NodeId {
+        use ensogl_theme::graph_editor as theme;
+        let parent_pos = self.model.get_node_position(parent).unwrap_or_default();
+        let styles     = StyleWatch::new(&self.model.app.display.scene().style_sheet);
+        let x_gap      = styles.get_number(theme::default_x_gap_between_nodes);
+        let y_gap      = styles.get_number(theme::default_y_gap_between_nodes);
```

Comment:
This is slow as well - this should be read in FRP by using StyleWatchFRP instead. Now, we are using a lot of string queries in hashmaps everytime we create a new node. When using FRP, this will be done only when theme changes.